app/jobs/remove_solr_collection_job.rb

```diff
@@ -0,0 +1,15 @@
+class RemoveSolrCollectionJob < ActiveJob::Base
+  # @param collection [String] the name of the collection
+  # @param connection_options [Hash] options for connecting to solr.
+  # @option connection_options [String] :url
+  # @option connection_options [String] :url
+  def perform(collection, connection_options)
+    solr_client(connection_options).get '/solr/admin/collections', params: { action: 'DELETE', name: collection }
+  end
+
+  private
+
+    def solr_client(connection_options)
+      RSolr.connect(connection_options)
```

Comment:
The Blacklight.default_index is mutable (changed by Account#switch). Since we're moving the solr call to a time after the account record has been deleted, we need to make it immutable.